pkg/inventory/web/client.go

```diff
@@ -435,13 +447,17 @@ func (r *Watch) ID() uint64 {
 }
 
 //
-// End the watch.
+// Repair the watch.
 func (r *Watch) Repair() (err error) {
 	status, err := r.reader.Repair()
 	if err != nil {
 		return
 	}
-	if status != http.StatusOK {
+	switch status {
+	case http.StatusOK:
+	case http.StatusNotFound:
+		r.End()
```

Comment:
Detecting the 404 here is needed to abort the watch `repair` if the resource no longer exits.